mobile/src/test/java/org/openhab/habdroid/model/OpenHAB2WidgetTest.java

```diff
@@ -1,67 +1,404 @@
+    }
+
+    @Test
+    public void testGetPeriode() throws Exception {
+        assertEquals("M", sut1.getPeriod());
+        assertEquals("D", sut2.getPeriod()); // Object has no period, should default to D
+    }
+
+    @Test
+    public void testGetStep() throws Exception {
+        assertEquals(1F, sut1.getStep());
+        assertEquals(0.1F, sut2.getStep());
+    }
+
+    @Test
+    public void testGetRefresh() throws Exception {
+        assertEquals(1000, sut1.getRefresh());
+        assertEquals(100, sut2.getRefresh()); // Min refresh is 100, object has set refresh to 10
+    }
+
+    @Test
+    public void testHasItem() throws Exception {
+        assertEquals(true, sut1.hasItem());
+        assertEquals(true, sut2.hasItem());
+    }
+
+    @Test
+    public void testHasLinkedPage() throws Exception {
+        assertEquals(true, sut1.hasLinkedPage());
+        assertEquals(false, sut2.hasLinkedPage());
+    }
+
+    @Test
+    public void testHasMappings() throws Exception {
+        assertEquals(true, sut1.hasMappings());
+        assertEquals(true, sut2.hasMappings());
+    }
+
+    @Test
+    public void testChildrenHasLinkedPages() throws Exception {
+        assertEquals(false, sut1.childrenHasLinkedPages());
+        assertEquals(false, sut2.childrenHasLinkedPages());
+        assertEquals(true, sut3.childrenHasLinkedPages());
+
+    }
+
+    @Test
+    public void testChildrenHasNonlinkedPages() throws Exception {
+        assertEquals(true, sut1.childrenHasNonlinkedPages());
+        assertEquals(false, sut2.childrenHasNonlinkedPages());
+
+    }
+
+    @Test
+    public void testGetColors() throws Exception {
+        // Doesn't seem to work via Android Studio
+        Integer i = new Integer(0);
+        assertEquals(i, sut1.getLabelColor());
+        assertEquals(i, sut1.getIconColor());
+        assertEquals(i, sut1.getValueColor());
+        assertEquals(i, sut2.getLabelColor());
+        assertEquals(i, sut2.getIconColor());
+        assertEquals(i, sut2.getValueColor());
+    }
+
+    @Test
+    public void testGetType() throws Exception {
+        assertEquals("Group", sut1.getType());
+        assertEquals("Group", sut2.getType());
+        assertEquals("Frame", sut3.getType());
+        assertEquals("Switch", sut3.getChildren().get(1).getType());
+        assertEquals("Group", sut3.getChildren().get(2).getType());
+    }
+
+    @Test
+    public void testGetLabel() throws Exception {
+        assertEquals("Group1", sut1.getLabel());
+        assertEquals("Group1", sut2.getLabel());
+        assertEquals("Dimmer [81 %]", sut3.getChildren().get(0).getLabel());
+    }
+
+    @Test
+    public void testGetMapping() throws Exception {
+        assertEquals("ON", sut1.getMapping(0).getCommand());
+        assertEquals("On", sut1.getMapping(0).getLabel());
+        assertEquals("abcäöüßẞèéóò\uD83D\uDE03", sut2.getMapping(0).getLabel());
+
+        IndexOutOfBoundsException e = null;
```

Comment:
See my comment above, where you used such kind of code. I would extract it as a separate test and use the expected parameter of the Test annotation.